### PR TITLE
Create cassette submodule

### PR DESCRIPTION
**Pull request by merwanehamadi**
### Background

Cassettes are too big, they need their own submodule

### Changes

- Created submodule
- add ability to update cassettes in ci pipeline
### Documentation

Screenshot 2023-05-25 at 6 38 34 PM

Screenshot 2023-05-25 at 6 38 27 PM

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes. 
- [x] I have run `black .` and `isort .` against my code to ensure it passes our linter.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,9 +2,11 @@ name: Python CI
 
 on:
   push:
-    branches: [ master ]
+    branches: [ master, ci-test*]
+    paths-ignore:
+      - 'tests/Auto-GPT-test-cassettes'
   pull_request_target:
-    branches: [ master, stable ]
+    branches: [ master, stable , ci-test*]
```
Comment (waynehamadi) [inline .github/workflows/ci.yml:9]: going to make testing easier in the future

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -2,9 +2,11 @@ name: Python CI
 
 on:
   push:
-    branches: [ master ]
+    branches: [ master, ci-test*]
+    paths-ignore:
+      - 'tests/Auto-GPT-test-cassettes'
```
Comment (waynehamadi) [inline .github/workflows/ci.yml:7]: without this, there is an infinite loop

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -71,6 +73,33 @@ jobs:
           fetch-depth: 0
           ref: ${{ github.event.pull_request.head.ref }}
           repository: ${{ github.event.pull_request.head.repo.full_name }}
+          submodules: true
+          token: ${{ secrets.PAT_REVIEW }}
+
+      - name: Use cassettes previously sent
```
Comment (waynehamadi) [inline .github/workflows/ci.yml:79]: only use the cassettes if you're working on a pull request and there is no conflict with master

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -93,3 +122,82 @@ jobs:
...
+          git fetch origin ${{ github.event.pull_request.base.ref }}:${{ github.event.pull_request.base.ref }} || echo "DIFF_EXISTS=false" >> $GITHUB_ENV
+          differences=$(git diff origin/$new_branch origin/${{ github.event.pull_request.base.ref }})
+    
+          if [ -n "$differences" ]; then
+            echo "DIFF_EXISTS=true" >> $GITHUB_ENV
+          else
+            echo "DIFF_EXISTS=false" >> $GITHUB_ENV
+          fi
+
+      - name: Apply or remove prompt change label and comment
```
Comment (waynehamadi) [inline .github/workflows/ci.yml:180]: inform when there is a prompt change so we catch it

File: .github/workflows/docker-ci.yml [hunk 5]
```diff
@@ -76,41 +76,44 @@ jobs:
...
-
-        echo "$test_output"
-
-        cat << $EOF >> $GITHUB_STEP_SUMMARY
-        # Tests $([ $test_failure = 0 ] && echo '✅' || echo '❌')
-        \`\`\`
-        $test_output
-        \`\`\`
-        $EOF
+      - name: Check out repository
```
Comment (waynehamadi) [inline .github/workflows/docker-ci.yml:79]: the indentation was bad

File: autogpt/llm/utils/__init__.py [hunk 6]
```diff
@@ -96,7 +96,7 @@ def _wrapped(*args, **kwargs):
                         user_warned = True
 
                 except APIError as e:
-                    if (e.http_status != 502) or (attempt == num_attempts):
+                    if (e.http_status not in [502, 429]) or (attempt == num_attempts):
```
Comment (waynehamadi) [inline autogpt/llm/utils/__init__.py:99]: some errors are 429 and yet they're not RateLimitError, it's an openai issue

**Review by k-boikov — APPROVED** [review]

**Pull request merged**